0000-catalog.md

```diff
@@ -0,0 +1,744 @@
+participants in a grid network.
+
+
+Any grid product added to a catalog (thus becoming a catalog_product) will need to 
+contain a reference fields to the original product (product_id, product_namespace, 
+and owner) and a catalog_id. A catalog product schema may also be defined to 
+required additional fields for that catalog_product. This example schema enforces 
+a _required_ ***status*** field and an _optional_ ***prices*** field.
+
+Example schema:
+```
+Schema(
+    name="Catalog Product",
+    description="A grid product that can be grouped with other products via grid catalog",
+    owner = "Target"
+    properties=[
+        PropertyDefinition(
+            name="status",
+            data_type=PropertyDefinition.DataType.ENUM,
+            description="The current state of the catalog product",
+            enum_options=["ACTIVE", "INACTIVE"],
+            required=True
+        ),
+        PropertyDefinition(
+            name="price",
+            data_type=PropertyDefinition.DataType.STRING,
+            description="The price of the catalog product"
+            required=False
+        ),
+        PropertyDefinition(
+            name="return_policy",
+            data_type=PropertyDefinition.DataType.STRING,
+            description="The return policy of the of the catalog product"
+            required=False
+        )])
+```
+
+In order to store a catalog product in global state, we will extend from the 
+definition of a grid product with several fixed fields, namely identifiers, as 
+well as the set of dynamic property values that will conform to the catalog 
+product schema definition.
+
+**Grid Product from the [Product RFC](https://github.com/hyperledger/grid-rfcs/blob/master/text/0005-product.md):**
+
+```
+message Product { 
+    enum ProductNamespace { 
+        UNSET_NAMESPACE = 0; 
+        GS1 = 1; 
+    }
+    
+    ProductNamespace product_namespace = 1; 
+    string product_id = 2; 
+    string owner = 3;
+    repeated PropertyValue properties = 4; 
+}
+```
+
+The catalog_product smart contract would then be responsible for validating the 
+properties field against the CatalogProduct schema at run-time.
+
+A catalog_product entry would reference Grid Product. In addition to 
+the properties defined in the CatalogProduct schema. The data model 
+would look like this:
+```
+CatalogProduct(
+    product_id="gtin", // Reference from Grid Product
+    product_namespace="GS1" // Reference from Grid Product
+    owner='Target', // Reference from Grid Product
+		catalog_id='e70fe351d1d096c'
+    properties=[
+        PropertyValue(
+            name="status",
+            data_type=PropertyDefinition.DataType.ENUM,
+            enum_value=1, # ACTIVE (default)
+        ),
+        PropertyValue(
+            name="price"
+            data_type=PropertyDefinition.DataType.STRING,
+            string_value="1-10:$50, 10-50:$40, 50+:$30"  # price windows
```

Comment:
Maybe make a comment here that real structured data for breakpoint discounts would likely be modeled in a schema, rather than this string example.